runelite-client/src/main/java/net/runelite/client/plugins/timetracking/farming/CompostTracker.java

```diff
@@ -0,0 +1,294 @@
+		PendingCompost pc = new PendingCompost(
+			Instant.now().plus(COMPOST_ACTION_TIMEOUT),
+			actionLocation,
+			targetPatch
+		);
+		pendingCompostActions.put(targetPatch, pc);
+	}
+
+	private boolean isCompostAction(MenuOptionClicked e)
+	{
+		switch (e.getMenuAction())
+		{
+			case WIDGET_TARGET_ON_GAME_OBJECT:
+				Widget w = client.getSelectedWidget();
+				assert w != null;
+				return compostItems.contains(w.getItemId()) || w.getId() == WidgetInfo.SPELL_LUNAR_FERTILE_SOIL.getPackedId();
+
+			case GAME_OBJECT_FIRST_OPTION:
+			case GAME_OBJECT_SECOND_OPTION:
+			case GAME_OBJECT_THIRD_OPTION:
+			case GAME_OBJECT_FOURTH_OPTION:
+			case GAME_OBJECT_FIFTH_OPTION:
+				return "Inspect".equals(e.getMenuOption());
+
+			default:
+				return false;
+		}
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage e)
+	{
+		if (e.getType() != ChatMessageType.GAMEMESSAGE && e.getType() != ChatMessageType.SPAM)
+		{
+			return;
+		}
+
+		CompostState compostUsed = determineCompostUsed(e.getMessage());
+		if (compostUsed == null)
+		{
+			return;
+		}
+
+		this.filterTimeouts();
+
+		pendingCompostActions.values()
+			.stream()
+			.filter(this::playerIsBesidePatch)
+			.findFirst()
+			.ifPresent(pc ->
+			{
+				setCompostState(pc.getFarmingPatch(), compostUsed);
+				pendingCompostActions.remove(pc.getFarmingPatch());
+			});
+	}
+
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged e)
+	{
+		switch (e.getGameState())
+		{
+			case LOGGED_IN:
+			case LOADING:
+				return;
+
+			default:
+				pendingCompostActions.clear();
+		}
+	}
+
+	private boolean playerIsBesidePatch(PendingCompost pendingCompost)
+	{
+		// find gameobject instance in scene
+		// it is possible that the scene has reloaded between use and action occurring so we use worldpoint
+		// instead of storing scene coords in the menuoptionclicked event
+		LocalPoint localPatchLocation = LocalPoint.fromWorld(client, pendingCompost.getPatchLocation());
+		if (localPatchLocation == null)
+		{
+			return false;
+		}
+
+		@Varbit int patchVarb = pendingCompost.getFarmingPatch().getVarbit();
+		Tile patchTile = client.getScene()
+			.getTiles()[client.getPlane()][localPatchLocation.getSceneX()][localPatchLocation.getSceneY()];
+		GameObject patchObject = null;
+		for (GameObject go : patchTile.getGameObjects())
+		{
+			if (go != null && client.getObjectDefinition(go.getId()).getVarbitId() == patchVarb)
+			{
+				patchObject = go;
+				break;
+			}
+		}
+		assert patchObject != null;
+
+		// player coords
+		Player p = client.getLocalPlayer();
+		int playerX = p.getWorldLocation().getX();
+		int playerY = p.getWorldLocation().getY();
+
+		// patch coords
+		int minX = patchObject.getWorldLocation().getX();
+		int minY = patchObject.getWorldLocation().getY();
+		int maxX = minX + patchObject.sizeX() - 1;
+		int maxY = minY + patchObject.sizeY() - 1;
+
+		// player should be within one tile of these coords
+		// todo corner detection?
```

Comment:
you can't interact with objects from corners, but maybe you can use spells from corners or further away? Have you checked that?

Reply:
The spell drags the player toward the patch and places you beside it before activating. I have not tried corners for casting yet.

Reply:
Reconfirmed, can't cast the spell from a distance or diagonally. The player is dragged toward the patch.